src/Domain.LinnApps/Reports/SpendsReportService.cs

```diff
@@ -0,0 +1,157 @@
+
+            // do I need to check the current ledger period as an upper limit? Not sure if we'd have stuff in the future
+            var supplierSpends = this.spendsRepository.FilterBy(
+                        x => x.LedgerPeriod >= previousYearStartLedgerPeriod && x.LedgerPeriod <= currentLedgerPeriod
+                             && (string.IsNullOrWhiteSpace(vm) || x.Supplier.VendorManager == vm))
+                    .ToList();
+
+            var vmName = "ALL";
+            if (!string.IsNullOrWhiteSpace(vm))
+            {
+                var vendorManager = this.vendorManagerRepository.FindById(vm);
+                vmName = $"{vm} - {vendorManager.Employee.FullName} ({vendorManager.UserNumber})";
+            }
+
+            var reportLayout = new SimpleGridLayout(
+                this.reportingHelper,
+                CalculationValueModelType.TextValue,
+                null,
+                $"Spend by supplier report for Vendor Manager: {vmName}. For this financial year and last, excludes factors & VAT.");
+
+            this.AddSupplierReportColumns(reportLayout);
+
+            var values = new List<CalculationValueModel>();
+
+            var distinctSupplierSpends = supplierSpends.DistinctBy(x => x.SupplierId).Select(
+                x => new SupplierSpend
+                         {
+                             BaseTotal = x.BaseTotal,
+                             LedgerPeriod = x.LedgerPeriod,
+                             Supplier = x.Supplier,
+                             SupplierId = x.SupplierId,
+                             MonthTotal =
+                                 supplierSpends.Where(
+                                         s => s.SupplierId == x.SupplierId && s.LedgerPeriod == currentLedgerPeriod)
+                                     .Sum(z => z.BaseTotal),
+                             YearTotal =
+                                 supplierSpends.Where(
+                                         s => s.SupplierId == x.SupplierId && s.LedgerPeriod >= yearStartLedgerPeriod)
+                                     .Sum(z => z.BaseTotal),
+                             PrevYearTotal = supplierSpends.Where(
+                                     s => s.SupplierId == x.SupplierId
+                                          && s.LedgerPeriod >= previousYearStartLedgerPeriod
+                                          && s.LedgerPeriod < yearStartLedgerPeriod)
+                                 .Sum(z => z.BaseTotal)
+                         }).OrderByDescending(x => x.PrevYearTotal).ThenByDescending(s => s.YearTotal).ThenByDescending(s => s.MonthTotal);
+
+            foreach (var supplier in distinctSupplierSpends)
+            {
+                this.ExtractDetailsForPartReport(values, supplier);
+            }
+
+            reportLayout.SetGridData(values);
+            var model = reportLayout.GetResultsModel();
+
+            return model;
+
```

Comment:
blank line

Reply:
Sorry, I hadn't looked through the files when I put this up, really should have. Wasn't on the ball last night. I'll need to check my stylecop too, I was getting warnings I'm sure but seem to have missed more than usual